lib/puppet/application/agent.rb

```diff
@@ -359,6 +359,12 @@ def main(daemon)
     daemon.start
   end
 
+  def parse_args(args)
```

Comment:
I'm not a fan of having a `parse_args` method that doesn't really parse the arguments, but instead ensures they are empty.

I think this really could simply be a one liner at the start of `setup`:

```ruby
raise ArgumentError, "The puppet agent command does not take parameters" unless command_line.args.empty?
```